controller/src/main/java/org/jboss/as/controller/PathAddress.java

```diff
@@ -150,6 +147,68 @@ public static PathAddress pathAddress(PathAddress parent, PathElement... element
         return pathAddress(list);
     }
 
+    public static PathAddress parseAddress(String address) throws IllegalArgumentException {
```

Comment:
This is actually parsing a CLI-style address, so the method name should reflect that; something like parseCLIStyleAddress. This names corresponds to the toCLIStyleString method.

The other style is HTTP style, where the address is part of a GET URL. That style uses '/' instead of '='.